### PR TITLE
Move SDK-specific cmake logic and headers out of bindgen-lib

### PR DESCRIPTION
**Pull request by RedBeard0531**
I think after these changes, the bindgen-lib dir now only has SDK-agnostic content. Once this is merged, the next step is to move the bindgen-lib directory to a core branch, then reshuffle the js submodule structure so that the core submodule is in `packages/realm/bindgen/vendor/realm-core`

NOTE TO REVIEWERS: I suggest reviewing 1 commit at a time.

### CODE REVIEW

File: packages/realm/bindgen/CMakeLists.txt
Comment (RedBeard0531) [inline packages/realm/bindgen/CMakeLists.txt:1]: This file was cut-paste extracted from vendor/bindgen-lib/CMakeLists.txt. The only changes were

* Bump `cmake_minimum_version()` to 3.17 (line31)
* remove two levels of `../../` from the `PACKAGE_ROOT_DIR` definition (line 67)
* added `add_subdirectory(vendor/bindgen-lib EXCLUDE_FROM_ALL)` (line 119)

Comment (RedBeard0531) [reply packages/realm/bindgen/CMakeLists.txt:1]: NB: this is only true for the second commit which did the extraction. The first commit made other changes. Look at it to see them.

**Review by kraenhansen — APPROVED** [review]
Fascinating cmake magic 🪄

**Comment by RedBeard0531** [comment]
> Fascinating cmake magic 🪄

I got a bit of guidance from the #cmake channel on cpplang slack. Thanks @mathstuf!

**Pull request merged**